### ISSUE TITLE
LEAST and GREATEST : TypeError: Object #<Object> has no method 'join'

### ISSUE THREAD

**Issue by gregaubert** [agershun/alasql#104]
When trying to use the __GREATEST__ or __LEAST__ functions with this exemple :

```javascript
var alasql = require('alasql');
var tab = [{a: 1, b:4}, {a:5, b:3}];
alasql('SELECT GREATEST(a, b) as test FROM ?', tab);
```
We get the error _<Object> has no method 'join' _ because we try to use join on the [arguments](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments) object, and it's not real array, so it got no method __join__.

It is possible to solve it by calling the __join__ method like this : 

```javascript
Array.prototype.join.call(arguments, ',')
```

But if I update alasql with this fix, then I get __NaN__ results instead of the numbers : 

```javascript
 alasql('SELECT GREATEST(a, b) as test FROM ?', tab); 
// result : [ { test: NaN }, { test: NaN } ]
```

And the result of the __join__ method called on the arguments in the __GREATEST__ method gives this : 
```javascript
console.log(Array.prototype.join.call(arguments, ','));
// result : p['default']['a'],p['default']['b']
```
Is this correct ? Or could the problem come from that ?

**Comment by agershun** [comment]
Mea culpa!

You can temporary fix it with:
```js
    alasql.stdlib.GREATEST = function(){
          return 'Math.max('+Array.prototype.slice.call(arguments).join(',')+')'
    };
    alasql.stdlib.LEAST = function(){
          return 'Math.min('+Array.prototype.slice.call(arguments).join(',')+')'
    };
```
See full example [in jsFiddle](http://jsfiddle.net/mrweh7wo/1/)

I will fix this bug and update the library tonight.

**Comment by agershun** [comment]
BTW: You missed brackets in ```alasql()``` data argument:
```js
    alasql('SELECT GREATEST(a, b) as test FROM ?', tab); // wring
    alasql('SELECT GREATEST(a, b) as test FROM ?', [tab]); // correct
```

**Comment by gregaubert** [comment]
Oh yeah so that's the reason for __NaN__ value. Okay my bad.

For the temporary fix, you don't need to _slice_ before the _join_, this works too : 

```javascript
    alasql.stdlib.GREATEST = function(){
          return 'Math.max('+Array.prototype.join.call(arguments, ',')+')'
    };
    alasql.stdlib.LEAST = function(){
          return 'Math.min('+Array.prototype.join.call(arguments, ',')+')'
    };
```

**Comment by agershun** [comment]
Wow! :+1:

**Comment by agershun** [comment]
Fixed in ```develop``` branch (this fix will be relased in version 0.0.43)
